### ISSUE TITLE
Disabling the MissingTypeAnnotation linting rule disabled the "Add inferred annotation" code action.

### ISSUE THREAD

**Issue by anagrius** [elm-tooling/elm-language-server#827]
There is an unfortunate interaction between the linting rule "MissingTypeAnnotation" and "Add inferred annotation" code action.

If you disable the linting rule, you cannot trigger the code action for top-level functions missing a type annotation.

I would be happy to make a PR fixing this. If we agree that there is no reason to disable the code action just because the linter does not report it as an error.

**Comment by razzeee** [comment]
Yeah, if that can be sanely done, without adding too much maintenance burden
